Translate graphql-api (#56)
gatsbyjs/gatsby-pl: docs/docs/graphql-api.md

tableOfContentsDepth: 2
---

> Ogromną przewagą Gatsby jest wbudowana warstwa danych która łączy wszystkie źródła danych jakie skonfigurujesz. Dane są zbierane w [czasie kompilacji](/docs/glossary#build) i automatycznie zostają zgromadzone w [schemat](/docs/glossary#schema) który definiuje w jaki sposób dane mogą być odpytywane przez twoją stronę.

Reviewer: Generalnie będę dawać zmiany dot. "naturalności" tłumaczenia. Samemu ciężko takie problemy znaleźć (a jeszcze trudniej rozwiązać), więc trzeba sobie pomagać :)

```suggestion
Ogromną przewagą Gatsby jest wbudowana warstwa danych która jest w stanie połączyć wiele źródeł danych. Dane zbierane są w [trakcie budowania](/docs/glossary#build) i zostają automatycznie zgromadzone w [schemat](/docs/glossary#schema) który definiuje w jaki sposób mogą one być pobierane przez twoją stronę.
```

> Ten dokument służy jako odniesienie do funkcji GraphQL wbudowanych w Gatsby, włącznie z metodami do zapytań i pozyskiwania danych, oraz dostosowywania GraphQL do potrzeb twojej strony.

Reviewer: ```suggestion
Jest to dokumentacja funkcjonalności GraphQL wbudowanych w Gatsby, włącznie z metodami do zapytań i pozyskiwania danych, oraz dostosowywania GraphQL do swoich potrzeb.
```

## Pierwsze kroki z GraphQL

> GraphQL jest dostępny w Gatsby bez potrzeby dodatkowej instalacji: schemat jest automatycznie wyciągany i tworzony kiedy uruchamiasz projekt `gatsby develop` lub `gatsby build`. Kiedy strona zostanie skompilowania, warstwę danych możesz [zbadać](/docs/running-queries-with-graphiql/) pod adresem: <http://localhost:8000/___graphql>

Reviewer: ```suggestion
GraphQL jest dostępny w Gatsby bez potrzeby dodatkowej instalacji: schemat jest automatycznie budowany kiedy uruchamiasz projekt (`gatsby develop` lub `gatsby build`). Kiedy strona zostanie zbudowana, warstwę danych możesz [ręcznie przeglądać](/docs/running-queries-with-graphiql/) pod adresem: <http://localhost:8000/___graphql>

## Pozyskiwanie danych

> Dane muszą być [pozyskane](/docs/content-and-data/) - lub dodane do schematu GraphQL - aby odpytać i wykorzystać je w stronach za pośrednictwem GraphQL. Gatsby wykorzystuje [wtyczki źródłowe](/plugins/?=gatsby-source) do pobierania danych.

Reviewer: Tutaj typowo konstrukcja z języka angielskiego ;)

```suggestion
Dane muszą zostać [dodane](/docs/content-and-data/) do schematu GraphQL, aby je odpytywać i wykorzystywać na stronie za pomocą GraphQL. Do pobierania danych, Gatsby wykorzystuje [wtyczki źródłowe](/plugins/?=gatsby-source).
```

> **Uwaga**: GraphQL nie jest wymagany: możesz wciąż [wykorzystywać Gatsby bez GraphQL](/docs/using-gatsby-without-graphql/).

Reviewer: ```suggestion
**Uwaga**: GraphQL nie jest wymagany: z Gatsby można [korzystać bez GraphQL](/docs/using-gatsby-without-graphql/).
```

> Aby pozyskać dane poprzez istniejącą wtyczkę, musisz zainstalować wszystkie potrzebne paczki. Ponadto musisz dodać wtyczkę do tablicy wtyczek w `gatsby-config` i jej opcjonalną konfigurację. Jeśli chcesz pozyskać dane z systemu plików do użytku z GraphQL, jak na przykład pliki Markdown, zdjęcia i inne, sprawdź [dokumentację pozyskiwania danych z systemu plików](/docs/sourcing-from-the-filesystem/) oraz [przepisy](/docs/recipes/sourcing-data).

Reviewer: ```suggestion
Aby pozyskać dane poprzez istniejącą wtyczkę, musisz zainstalować wszystkie potrzebne paczki i dodać wtyczkę (oraz opcjonalnie jej konfigurację) do tablicy wtyczek w `gatsby-config`. Dane takie jak zdjęcia, pliki Markdown, itp. można pozyskiwać z systemu plików. Po więcej informacji, sprawdź [dokumentację pozyskiwania danych z systemu plików](/docs/sourcing-from-the-filesystem/) oraz [przepisy](/docs/recipes/sourcing-data).
```

Aby uzyskać instrukcje dotyczące instalacji paczek z npm, zajrzyj do instrukcji w dokumentacji [jak korzystać z wtyczki](/docs/using-a-plugin-in-your-site/).

> Możesz również [stworzyć niestandardowe wtyczki](/docs/creating-plugins/) aby pasowały do twoich przypadków użycia i wyciągaj dane jak tylko chcesz.

Reviewer: ```suggestion
Możesz również [tworzyć niestandardowe wtyczki](/docs/creating-plugins/) dopasowane do twoich potrzeb, które mogą pobierać dane w dowolny sposób.
```

> ## Odpytywanie w komponentach i hooki

Reviewer: ```suggestion
## Komponenty i hooki służące do wykonywania zapytań
```

Dane mogą być odpytywane w stronach, komponentach, oraz w pliku `gatsby-node.js`, wykorzystując jedną z poniższych możliwości:

- Komponent `pageQuery`
- Komponent `StaticQuery`
- Hook `useStaticQuery`

> **Uwaga**: Ze względu na to jak Gatsby przetwarza zapytania GraphQL, nie możesz mieszać zapytań o strony i statycznych zapytać w tym samym pliku. Nie możesz również mieć wielokrotnych zapytań o strony i zapytań o strony w jednym pliku.

Reviewer: Tu nie chodzi o "zapytania o strony", a o specjalny typ zapytania zarezerwowany dla stron :)

```suggestion
**Uwaga**: Ze względu na to jak Gatsby przetwarza zapytania GraphQL, nie możesz mieszać zapytań stron i zapytań statycznych wewnątrz plik. Nie możesz również umieszczać wielu zapytań stron wewnątrz jednego pliku.
```

Author: Tak dokładnie, nie wiedziałem do końca jak "page query" przetłumaczyć :P

Author: Drobna poprawka, 'wewnątrz pliku' powinno być, poprawię już

> Po więcej informacji na temat komponentów strony lub innych w odniesieniu do zapytań, sprawdź przewodnik w dokumentacji [budowanie z komponentami](docs/building-with-components/#how-does-gatsby-use-react-components)

Reviewer: ```suggestion
Po więcej informacji na wykorzystywania przez Gatsby komponentów stron i nie tylko, sprawdź sekcję [budowanie z komponentami](docs/building-with-components/#how-does-gatsby-use-react-components).
```

### `pageQuery`

> ===========================================

Reviewer: Czo? :D

Author: Nie mam pojęcia 😅

> `pageQuery` jest wbudowanym komponentem który pobiera informacje z warstwy danych w stronach Gatsby. Na jednej stronie możesz mieć jedno zapytanie. Może ono przyjmować argumenty GraphQL dla zmiennych w zapytaniach.

Reviewer: ```suggestion
`pageQuery` jest wbudowanym komponentem który pobiera informacje z warstwy danych w stronach Gatsby. Na jednej stronie możesz mieć jedno zapytanie. Może ono przyjmować argumenty dla zmiennych w zapytaniach.
```

> [Strona jest utworzona w Gatsby](/docs/page-creation/) przez dowolny React komponent znajdujący się w folderze `src/pages` lub poprzez wywołanie akcji `createPage` oraz użycie komponentu w opcjach `createPage` - znaczy to, że `pageQuery` nie zadziała w dowolnym komponencie, ale tylko w tych które spełniają te kryteria.

Reviewer: ```suggestion
[Strona jest utworzona w Gatsby](/docs/page-creation/) z dowolnego komponentu React znajdującego się w folderze `src/pages` lub poprzez wywołanie akcji `createPage` oraz użycie komponentu w opcjach `createPage` - znaczy to, że `pageQuery` nie zadziała w dowolnym komponencie, ale tylko w tych które spełniają te kryteria.
```

> Zapoznaj się również z [przewodnikiem jak odpytywać dane w stronach poprzez zapytanie strony](/docs/page-query/)

Reviewer: ```suggestion
Zapoznaj się również z [przewodnikiem dotyczącym pobierania danych w stronach poprzez zapytania stron](/docs/page-query/)

`
```

**Uwaga**: Zapytanie eksportowane jako `const` wcale nie musi być nazwane `pageQuery`. Ważniejszym aspektem jest to, że Gatsby szuka w pliku wyeksportowanego string'a `graphql`.

> #### Zwroty

Reviewer: ```suggestion
#### Zwrócone wartości
```

> Kiedy zapytanie jest użyte w pliku komponentu strony, zapytanie strony zwraca obiekt `data` który automatycznie jest przekazywany do komponentu jako atrybut.

Reviewer: ```suggestion
Kiedy zapytanie jest użyte w pliku komponentu strony, zapytanie strony zwraca obiekt `data` zostaje przekazany do komponentu jako prop.
```

Author: Tutaj podobnie :)

### `StaticQuery`

> StaticQuery jest wbudowanym komponentem który pobiera dane z warstwy danych Gatsby dla komponentów które nie są stronami, jak np. header, nawigacja czy dowolny inny komponent użyty jako dziecko.

Reviewer: ```suggestion
StaticQuery jest wbudowanym komponentem który pobiera dane z warstwy danych Gatsby dla komponentów które nie są stronami, jak np. nagłówek, nawigacja czy dowolny inny komponent użyty jako dziecko.
```

Możesz użyć tylko jedno `StaticQuery` na stronę: aby uwzględnić potrzebne dane z wielu źródeł, możesz użyć jednego zapytania z wieloma [głównymi polami](/docs/graphql-concepts/#query-fields). Komponent ten nie może przyjmować zmiennych jako argumenty.

> Zapoznaj się również z [przewodnikiem jak odpytywać dane w komponentach wykorzystując zapytanie statyczne](/docs/static-query/).

Reviewer: ```suggestion
Zapoznaj się również z [przewodnikiem dotyczącym odpytywania danych w komponentach wykorzystując zapytanie statyczne](/docs/static-query/).
```

#### Parametry

> Komponent `StaticQuery` przyjmuje dwie wartości jako atrybuty w JSX:

Reviewer: ```suggestion
Komponent `StaticQuery` przyjmuje dwie wartości jako propsy w JSX:
```

Author: Tutaj podobnie :)

/>
```

#### Zwroty

> Komponent StaticQuery zwraca `data` w jako atrybut `render`:

Reviewer: ```suggestion
Komponent StaticQuery zwraca `data` argument w `render` prop:
```

Author: Tutaj podobnie (prop na atrybut)

### `useStaticQuery`

> Hook `useStaticQuery` może być użyty podobnie do `StaticQuery` w dowolnym komponencie lub stronie, ale nie wymaga użycia komponentu i atrybutu render.

Reviewer: ```suggestion
Hook `useStaticQuery` może być użyty w podobny sposób do `StaticQuery` w dowolnym komponencie lub stronie, ale nie wymaga użycia dodatkowego komponentu z render propem.
```

Author: Sprawdzałem polską dokumentację reacta i tam propsy przetłumaczyli na atrybuty, może warto pozostać przy tym tłumaczeniu?

Reviewer: Sprawdzę jeszcze raz jak tłumaczyliśmy tutaj, ale na 90% tłumaczyliśmy tutaj na propsy. Wydaje mnie się to bardziej naturalne niż "atrybuty", chociaż argument z dokumentacją Reacta ma jak najbardziej sens

Reviewer: Sprawdziłem, w innych tłumaczeniach jest `props` z polską odmianą :)

Author: Okej, też mi się wydaje że jest to bardziej zrozumiałe niż atrybut, aczkolwiek zastanawiałem się jak bardzo chcemy być zgodni z innymi dokumentacjami :D, to już wszystko jasne :)

> Ponieważ jest to React hook, [zasady korzystania z hooków](https://pl.reactjs.org/docs/hooks-rules.html) muszą zostać zachowane, więc będziesz potrzebował wersji React i ReactDOM 16.8.0 lub wyższej. Wobec tego to jak zapytania obecnie działają w Gatsby, tylko jedna instancja `useStaticQuery` jest wspierana dla danego pliku.

Reviewer: ```suggestion
Ponieważ jest to React hook, [zasady korzystania z hooków](https://pl.reactjs.org/docs/hooks-rules.html) muszą zostać zachowane, więc będziesz potrzebować wersji React i ReactDOM 16.8.0 lub wyższej. Wobec tego to jak zapytania obecnie działają w Gatsby, tylko jedna instancja `useStaticQuery` jest wspierana dla danego pliku.
```

> Zapoznaj się również z [poradnikiem jak odpytywać dane w komponentach z użyciem useStaticQuery](/docs/use-static-query/)

Reviewer: ```suggestion
Zapoznaj się również z [poradnikiem dotyczącym odpytywania danych w komponentach z użyciem useStaticQuery](/docs/use-static-query/)
```

)
```

## Struktora zapytania

> Zapytania są pisane w takim samym kształcie w jakim mają być zwracane dane. Sposów w jaki pozyskujesz dane określi nazwy pól, w odniesieniu do których możesz wysyłać zapytania, na podstawie węzłów które dodają do schematu GraphQL.

Reviewer: Trochę nie po polsku ;)

```suggestion
Zapytania są pisane w takim samym kształcie w jakim mają być zwracane dane. Umieszczone dane określą nazwy pól do których możesz tworzyć zapytania, bazując na node'ach które dodają do schematu GraphQL.
```

Aby lepiej zrozumieć działanie części zapytania, zapoznaj się z [przewodnikiem koncepcyjnym](/docs/graphql-concepts/#understanding-the-parts-of-a-query)

### Argumenty w zapytaniach GraphQL

Zapytania GraphQL mogą przyjmować argumenty które będą definiować jakie dane będą zwracane. Logika dla tych argumentów jest wewnęterznie obsługiwana przez Gatsby. Argumenty mogą być przekazywane do pól na dowolnym szczeblu zapytania.

> Różne węzły mogą przyjmować różne argumenty nw zależności od natury węzła

Reviewer: ```suggestion
Różne node'y w zależności od ich budowy, przyjmować mogą różne argumenty
```

> Argumenty które możesz przekazać do kolekcji(jak tablice lub długie listy dancyh - np. `allFile` czy `allMdx`) to:

Reviewer: ```suggestion
Argumenty które możesz przekazać do kolekcji (jak tablice lub długie listy danych - np. `allFile` czy `allMdx`) to:
```

- [`filter`](/docs/graphql-reference#filter)
- [`limit`](/docs/graphql-reference#limit)
- [`sort`](/docs/graphql-reference#sort)
- [`skip`](/docs/graphql-reference#skip)

Argumenty które możesz przekazać do pola `date` to:

- [`formatString`](/docs/graphql-reference#dates)
- [`locale`](/docs/graphql-reference#dates)

Argumenty które możesz przekazać do pola `excerpt` to:

- [`pruneLength`](/docs/graphql-reference#excerpt)
- [`truncate`](/docs/graphql-reference#excerpt)

### Operacje zapytań Graphql

Inne wbudowane konfiguracje mogą być użyte w zapytaniach

- [`Alias`](/docs/graphql-reference#alias)
- [`Group`](/docs/graphql-reference#group)

Jeśli chcesz zobaczyć przykłady, zapoznaj się z [recepturami zapytań](/docs/recipes/querying-data) i [przewodnikiem po opcjach zapytań GraphQL](/docs/graphql-reference/).

## Fragmenty zapytań

> Fragmenty pozwalają Ci na re-używanie części zapytań GraphQL. Pozwalają one również na dzielenie skomplikowanych zapytań na mniejsze, łatwiejsze w zrozumieniu komponenty.

Reviewer: ```suggestion
Fragmenty pozwalają Ci na ponowne używanie części zapytań GraphQL. Pozwalają one również na dzielenie skomplikowanych zapytań na mniejsze, łatwiejsze w zrozumieniu komponenty.
```
